### PR TITLE
[MDS-6082] Implement Caching.

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[MDS-6082](https://bcmines.atlassian.net/browse/MDS-6082)

_Why are you making this change? Provide a short explanation and/or screenshots_
Implemented response caching for LLM response.

### CODE REVIEW

**Review by taraepp — DISMISSED** [review]
Interesting that I can see your added test but sonarcloud thinks there's 0% coverage (probably need to fix the configuration there in the future)

LGTM!

File: services/permits/app/permit_conditions/pipelines/CachedAzureOpenAIChatGenerator.py [hunk 1]
```diff
@@ -64,33 +69,71 @@ def fetch_result(self, messages, generation_kwargs):
...
+            logger.info("cached_result: %s", cached_result)
+            res = {"replies": [ChatMessage(content=cached_result["hits"][0].content,
+                                           name=cached_result["hits"][0].meta["name"],
+                                           role=cached_result["hits"][0].meta["role"],
+                                           meta=cached_result["hits"][0].meta)]}
+        if not existing_reply_found:
             if DEBUG_MODE:
-                with open(f"app/cache/{cache_key}.pickle", "wb") as f:
-                    pickle.dump(res, f, protocol=pickle.HIGHEST_PROTOCOL)
+                try:
```
Comment (simensma-fresh) [inline services/permits/app/permit_conditions/pipelines/CachedAzureOpenAIChatGenerator.py]: Can probably get rid of the `pickle` cache now that we're using Elasticsearch

**Review by simensma-fresh — CHANGES_REQUESTED** [review]

**Review by matbusby-fw — DISMISSED** [review]
Once Simen's comment is resolved it looks good to me!

File: services/permits/app/permit_conditions/pipelines/CachedAzureOpenAIChatGenerator.py (hunk 1 above)
Comment (henryoforeh-dev) [reply services/permits/app/permit_conditions/pipelines/CachedAzureOpenAIChatGenerator.py]: Updated to remove all logic related to `pickle` cache.

**Review by matbusby-fw — APPROVED** [review]

**Review by simensma-fresh — APPROVED** [review]

**Pull request merged**